### PR TITLE
Update preferences views and add tests

### PR DESCRIPTION
**Pull request by cehsu**
## Description
These changes resolve the following issues:
- [#15219](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/15219)
- [#15220](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/15220)

TODO:
~- add all finalized content (in progress) (will address in subsequent PR)~
~- update formation version (related PR: https://github.com/department-of-veterans-affairs/design-system/pull/207)~
~- review formation upgrade changes~

## Testing done
- unit tests created

## Screenshots

No preferences selected:
image

Selected preference:
image

Selected preference removing view:
image

Set Preferences loading view:
image

## Acceptance criteria
- [x] Update preference views and content

## Definition of done
~- [ ] Events are logged appropriately~
- [x] Documentation has been updated, if applicable (in this case, the design system)
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by Samara-Strauss** [comment]
@cehsu screenshots are looking good! Some updates now that I am seeing things in code:

1. Can you please change the initial "Remove" button (in the screenshot above, the one next to the Health Care heading) to a link style? I was thinking of something like "Delete" style we use on some of the profile fields:

screen shot 2018-12-03 at 10 36 28 am

If that doesn't work for whatever reason, we can modify it to the secondary button style.

2. Can you please change the "Find VA Benefits" button to the secondary button style?
3. I added a comment re: the FVB page on aligning the checkboxes and benefit titles to #15220. Let me know your thoughts on that as well.

Also, I am not seeing these updates yet in staging — should I be?

Otherwise, things are looking great! Exciting to see this come to life 😃

File: src/applications/personalization/preferences/helpers.jsx
Comment (erikphansen) [inline src/applications/personalization/preferences/helpers.jsx]: we probably don't need the wrapper `div`

File: src/applications/personalization/preferences/helpers.jsx
Comment (erikphansen) [inline src/applications/personalization/preferences/helpers.jsx]: we probably don't need the wrapper `div`

**Comment by cehsu** [comment]
Thanks @samara-strauss, yes, we can use the delete button instead. I will also align the checkboxes. These changes will be up in staging once this PR is merged.

File: src/applications/personalization/preferences/tests/components/PreferenceOption.unit.spec.jsx
Comment (erikphansen) [inline src/applications/personalization/preferences/tests/components/PreferenceOption.unit.spec.jsx]: I think this is the first time I've seen component tests that look for specific markup like this. And it feels pretty fragile. Maybe in this case we could just check to make sure that the data from the passed-in `item` appears in the rendered component? Just a thought!

**Review by erikphansen — APPROVED** [review]
Had a few thoughts but generally looks good!

**Comment by Samara-Strauss** [comment]
> Thanks @Samara-Strauss, yes, we can use the delete button instead.

@cehsu to be clear, the copy should still say "Remove," but we can use the link style instead of the button style.

File: src/applications/personalization/preferences/helpers.jsx
Comment (cehsu) [reply src/applications/personalization/preferences/helpers.jsx]: Good call, will remove.

File: src/applications/personalization/preferences/helpers.jsx
Comment (cehsu) [reply src/applications/personalization/preferences/helpers.jsx]: It might be hard to notice, but the div is necessary in this case, as there are two lists.

File: src/applications/personalization/preferences/helpers.jsx
Comment (cehsu) [reply src/applications/personalization/preferences/helpers.jsx]: Good call.

File: src/applications/personalization/preferences/tests/components/PreferenceOption.unit.spec.jsx
Comment (cehsu) [reply src/applications/personalization/preferences/tests/components/PreferenceOption.unit.spec.jsx]: I should have added a comment that this is an experiment to simulate jest snapshot testing. I will replace with more conventional tests.

File: src/applications/personalization/preferences/containers/PreferencesWidget.jsx
Comment (cehsu) [reply src/applications/personalization/preferences/containers/PreferencesWidget.jsx]: Good question, how about we create a remove action instead so that the logic is not handled in the component?

Comment (erikphansen) [reply src/applications/personalization/preferences/containers/PreferencesWidget.jsx]: sooo... I don't really have a strong preference. I don't mind handling the logic here. And I guess since `setPreference` is just a redux action creator we probably don't need to worry about error handling like we would with something making an API call.

**Comment by cehsu** [comment]
Thanks @Samara-Strauss, kindly refer to these updated screenshots:

[Aligned checkboxes](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/15220#issuecomment-443751770):
image

[Updated remove content](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/15219#issuecomment-443765156):
image

[Remove button style](https://github.com/department-of-veterans-affairs/vets-website/pull/9186#issuecomment-443754121) & [Preferences saved confirmation](https://github.com/department-of-veterans-affairs/vets.gov-team/issues/15220#issuecomment-443863849):
image

**Comment by erikphansen** [comment]
@cehsu I know we want a +1 from @Samara-Strauss, but since she's out today and I'm building on this work I think we should merge it.

**Comment by cehsu** [comment]
Sounds good @erikphansen, it looks like this is a bit of a version bump for formation, so we'll probably want to test things out in staging a bit.

**Comment by Samara-Strauss** [comment]
Screenshots are looking great! Can't get into staging right now but will check it out there when I can.

**Comment by Samara-Strauss** [comment]
Screenshots are looking great! Can't get into staging right now but will check it out there when I can.

**Comment by Samara-Strauss** [comment]
Screenshots look good! Can't get onto staging right now but will check out these updates when I can.

**Comment by Samara-Strauss** [comment]
@cehsu can you please make the homelessness alert dismissible?

**Comment by erikphansen** [comment]
@Samara-Strauss and after dismissing if you shouldn't see it again the next time you come to the page?

**Comment by Samara-Strauss** [comment]
Also, @cehsu, can we tighten up this gap? It looks like a weird amount of space between the Find VA Benefits heading and the first block on the page:

screen shot 2018-12-06 at 2 56 27 pm

(Also, if this is not the right PR for this feedback, let me know where I can drop this)

**Comment by Samara-Strauss** [comment]
> @Samara-Strauss and after dismissing if you shouldn't see it again the next time you come to the page?

@erikphansen no, I would imagine you shouldn't see it again _unless_ you select another benefit with which it's associated. But if that gets complicated/weird, let me know.

**Comment by Samara-Strauss** [comment]
Two more things:

1. I noticed that the FVB selection page is missing the "I want to:" heading that sets up the checklist. Can you add that in? Here's a screenshot from the mockup:

screen shot 2018-12-06 at 4 15 08 pm

2. It looks to me like there is more padding on the bottom of each benefit block than the top. Is this because of how things shake out once they're in code, or is it possible to reduce the padding on the bottom of these boxes so it's more even with padding we're using at the top?

screen shot 2018-12-06 at 4 17 39 pm

**Comment by Samara-Strauss** [comment]
Another thing! These CTAs should be green:

screen shot 2018-12-06 at 4 21 27 pm

**Comment by erikphansen** [comment]
@cehsu @Samara-Strauss  I fixed the button color as part of my current work

**Comment by erikphansen** [comment]
@Samara-Strauss I also added the "I want to:" H3 header.

Can we put a pin in the padding/margin massaging until the content and functionality are more ironed out?

**Comment by Samara-Strauss** [comment]
> I fixed the button color as part of my current work

Awesome, will check it out!

> Can we put a pin in the padding/margin massaging until the content and functionality are more ironed out?

Totally, we can wait until the content is implemented.

**Comment by Samara-Strauss** [comment]
> I fixed the button color as part of my current work

Looks like this hasn't merged with staging yet. I'll keep an eye out for it when it does!

**Comment by Samara-Strauss** [comment]
Hey guys — Wanted to follow up on some stuff that I don't see on staging but I thought would have been merged by now:

1. ["I want to" copy](https://github.com/department-of-veterans-affairs/vets-website/pull/9186#issuecomment-445033249)
2. [Application CTAs should be green](https://github.com/department-of-veterans-affairs/vets-website/pull/9186#issuecomment-445034280)

Any idea why this stuff hasn't shown up yet?

Also, now that copy has been finalized, can we revisit some of the spacing issues I mentioned [here](https://github.com/department-of-veterans-affairs/vets-website/pull/9186#issuecomment-445009116) and [here](https://github.com/department-of-veterans-affairs/vets-website/pull/9186#issuecomment-445033249)?

**Comment by erikphansen** [comment]
Hey @Samara-Strauss!

Those two items were added to the big branch that Claire and I have been working on the past couple of weeks. But I'll move those to [this PR](https://github.com/department-of-veterans-affairs/vets-website/pull/9255) so you can see it on staging ASAP!

And, yeah, doing some CSS massaging to adjust the padding might be a nice change of pace for me later today.

**Comment by Samara-Strauss** [comment]
> Those two items were added to the big branch that Claire and I have been working on the past couple of weeks. But I'll move those to this PR so you can see it on staging ASAP!

Saw these updates on staging with the latest push. Everything looks great!!

**Pull request merged**
